### PR TITLE
Allow offline usage of TestNG

### PR DESCRIPTION
**Pull request by krmahadevan**
If the user is having network connectivity 
issues, today TestNG aborts execution.

For e.g., when there’s no internet connectivity
TestNG fails with “UnknownHostException” and thus
prevents users from using TestNG in offline mode.

Fixed this by falling back to using the embedded DTD
if we fail to read the one on the TestNG website.

### CODE REVIEW

**Review by juherr — CHANGES_REQUESTED** [review]
I think the embedded dtd should be the first to be used and I don't see any reason the check on the internet first.

BTW, you cannot force `testng-1.0.dtd` for any publicId/systemId like the current code is not forcing a specific url.

**Comment by krmahadevan** [comment]
@juherr - Fixed and pushed. Please take a look. It actually struck me that I was trying to solve a different problem only today. Current implementation always falls back to loading DTD from jar as long as its URL matches one of the following:

1. https://beust.com/testng/testng-1.0.dtd
2. https://testng.org/testng-1.0.dtd

Now incase a user ends up trying to refer to a DTD that is hosted elsewhere and/(or) just wants to use a DTD thats available in their classpath we should be supporting it (by first reading from classpath and then falling back to the URL they specify). 
It turns out that I was actually trying to fix this usecase. Got a bit confused in between. Now all is well. Please take another look.

File: src/main/java/org/testng/xml/TestNGContentHandler.java [hunk 1]
```diff
@@ -59,17 +60,24 @@
...
 
-      String newUrl = conn.getHeaderField("Location");
-      conn = (HttpURLConnection) new URL(newUrl).openConnection();
+        String newUrl = conn.getHeaderField("Location");
+        conn = (HttpURLConnection) new URL(newUrl).openConnection();
+      }
+      stream = conn.getInputStream();
     }
-    return new InputSource(conn.getInputStream());
+    return new InputSource(Objects.requireNonNull(stream, "Failed to load TestNG DTD from " + systemId));
```
Comment (juherr) [inline src/main/java/org/testng/xml/TestNGContentHandler.java:80]: Just remove the `TestNG` mention from the String

**Review by juherr — APPROVED** [review]

File: src/main/java/org/testng/xml/TestNGContentHandler.java (hunk 1 above)
Comment (krmahadevan) [reply src/main/java/org/testng/xml/TestNGContentHandler.java]: Fixed

**Pull request merged**